platforms/core-runtime/base-services/src/main/java/org/gradle/util/Path.java

```diff
@@ -312,4 +314,41 @@ public Path takeFirstSegments(int n) {
         }
         return new Path(Arrays.copyOfRange(segments, 0, Math.min(n, segmentCount())), absolute);
     }
+
+    /**
+     * Iterate over all ancestors of this path, starting with the root path (if absolute) or the first segment (if relative) and not including this path.
+     *
+     * <p>
+     * For example, the path {@code :a:b:c} has the ancestors {@code :}, {@code :a}, and {@code :a:b}.
+     * The path {@code a:b:c} has the ancestors {@code a}, and {@code a:b}.
+     * </p>
+     *
+     * @since 8.13
+     */
+    @Incubating
+    public Iterable<Path> ancestors() {
```

Comment:
Do we think this API provides value to our users? If so what is the use case?

If not I would avoid adding this method to public API

Reply:
I don't know if there's much value, but I'm not sure where else to put it. Do we have an existing class for useful Path methods that aren't public?

Reply:
I don't think so. I haven't seen where this is used but can we just put it in a static method there?

I think we should be very careful when adding public APIs, and only do so when explicitly necessary